test/core/verbencoding.js

```diff
@@ -0,0 +1,134 @@
+describe("Core htmx AJAX Verbs with json encoding", function() {
+    beforeEach(function () {
+        this.server = makeServer();
+        clearWorkArea();
+
+    });
+    afterEach(function () {
+        this.server.restore();
+        clearWorkArea();
+        // if we don't set this, then all the following test MAY 
+        // inherit incorrect encoding, and fail. 
+        htmx.config.defaultEncoding="xml";
+    });
+
+    it('handles basic post properly', function () {
+        var jsonResponseBody = JSON.stringify({});
+        this.server.respondWith("POST", "/test", jsonResponseBody);
+        var div = make('<div hx-post="/test encoding:json">click me</div>');
+        div.click();
+        this.server.respond();
+        this.server.lastRequest.response.should.equal("{}");
+    })
+
+    it('handles basic put properly', function () {
+        var jsonResponseBody = JSON.stringify({});
+        this.server.respondWith("PUT", "/test", jsonResponseBody);
+        var div = make('<div hx-put="/test encoding:json">click me</div>');
+        div.click();
+        this.server.respond();
+        this.server.lastRequest.response.should.equal("{}");
+    })
+
+    it('handles basic patch properly', function () {
+        var jsonResponseBody = JSON.stringify({});
+        this.server.respondWith("PATCH", "/test", jsonResponseBody);
+        var div = make('<div hx-patch="/test encoding:json">click me</div>');
+        div.click();
+        this.server.respond();
+        this.server.lastRequest.response.should.equal("{}");
+    })
+
+    it('handles basic delete properly', function () {
+        var jsonResponseBody = JSON.stringify({});
+        this.server.respondWith("DELETE", "/test", jsonResponseBody);
+        var div = make('<div hx-delete="/test encoding:json">click me</div>');
+        div.click();
+        this.server.respond();
+        this.server.lastRequest.response.should.equal("{}");
+    })
+
+    it('handles post with form parameters', function () {
+
+        this.server.respondWith("POST", "/test", function (xhr) {
+            var values = JSON.parse(xhr.requestBody);
+            values.should.have.keys("username","password");
+        });
+
+        var html = make('<form hx-post="/test encoding:json" > ' + 
+            '<input type="text"  name="username" value="joe"> ' +
+            '<input type="password"  name="password" value="123456"> ' +
+        '<button  id="btnSubmit">Submit</button> ');
+
+        byId("btnSubmit").click();
+        this.server.respond();
+    })
```

Comment:
And maybe add a response value based on the submitted values to test here to confirm the round trip?

Reply:
Added:

```
            var ans = { "passwordok": values["password"] == "123456"};
            xhr.respond(200, {}, JSON.stringify(ans));
```

and

```
     this.server.lastRequest.response.should.equal('{"passwordok":true}');
```
to 5 tests.